src/lib/core/WeaveExchangeMgr.cpp

```diff
@@ -666,7 +666,7 @@ void WeaveExchangeManager::DispatchMessage(WeaveMessageInfo *msgInfo, PacketBuff
 
     WeaveLogRetain(ExchangeManager, "Msg %s %08" PRIX32 ":%d %d %016" PRIX64 " %04" PRIX16 " %04" PRIX16 " %ld MsgId:%08" PRIX32,
                    "rcvd", exchangeHeader.ProfileId, exchangeHeader.MessageType,
-                   (int)msgBuf->DataLength(), msgInfo->SourceNodeId, msgCon->LogId(), exchangeHeader.ExchangeId,
+                   (int)msgBuf->DataLength(), msgInfo->SourceNodeId, msgCon == NULL ? -1 : msgCon->LogId(), exchangeHeader.ExchangeId,
```

Comment:
Stylistically, recommend putting the ternary comparison in parenthesis:

```suggestion
                   (int)msgBuf->DataLength(), msgInfo->SourceNodeId, ((msgCon == NULL) ? -1 : msgCon->LogId()), exchangeHeader.ExchangeId,
```